flink/README.md

```diff
@@ -0,0 +1,81 @@
+# Agent Check: flink
+
+## Overview
+
+This check monitors [Flink][1]. Datadog collects Flink metrics through Flink's
+[Datadog HTTP Reporter][4], which uses [Datadog's HTTP API][2].
+
+## Setup
+
+### Installation
+
+The Flink check is included in the [Datadog Agent][3] package.
+No additional installation is needed on your server.
+
+### Configuration
+
+1. Configure the [Datadog HTTP Reporter][4] in Flink.
+
+    Copy `<FLINK_HOME>/opt/flink-metrics-datadog-1.9.2.jar` into your `<FLINK_HOME>/lib` folder.
+    In your `<FLINK_HOME>/conf/flink-conf.yaml`, add these lines, replacing `<API_KEY>` with your Datadog API key:
+
+    ```yaml
+    metrics.reporter.dghttp.class: org.apache.flink.metrics.datadog.DatadogHttpReporter
+    metrics.reporter.dghttp.apikey: <API_KEY>
+    ```
+
+2. Re-map system scopes in your `<FLINK_HOME>/conf/flink-conf.yaml`.
+
+    ```yaml
+    metrics.scope.jm: flink.jobmanager
+    metrics.scope.jm.job: flink.jobmanager.job
+    metrics.scope.tm: flink.taskmanager
+    metrics.scope.tm.job: flink.taskmanager.job
+    metrics.scope.task: flink.task
+    metrics.scope.operator: flink.operator
+    ```
+
+    **Note**: The system scopes must be remapped for your Flink metrics to be supported, otherwise they will be submitted as custom metrics.
+
+3. Configure additional [tags][4] in `<FLINK_HOME>/conf/flink-conf.yaml`.
+    Here is an example of custom tags:
+
+    ```yaml
+    metrics.reporter.dghttp.tags: <KEY1>:<VALUE1>, <KEY1>:<VALUE2>
+    ```
+
+    **Note**: By default, any variables in metric names will be sent as tags, so there is no need to add custom tags for `job_id`, `task_id`, etc.
+
+4. Restart Flink to start sending your Flink metrics to the Agent.
```

Comment:
There seems to be no configuration required for the Agent, are we using DogStatsD or something?

Reply:
This is a tile only integration